DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -0,0 +1,422 @@
+using OpenDreamShared.Compiler.DM;
+using OpenDreamShared.Dream;
+using OpenDreamShared.Dream.Procs;
+using System;
+using System.Collections.Generic;
+
+namespace DMCompiler.DM.Visitors {
+    class DMVisitorExpression : DMASTVisitor {
+        DMObject _dmObject { get; }
+        DMProc _proc { get; }
+        DreamPath? _inferredPath { get; }
+        internal DMExpression Result { get; private set; }
+
+        internal DMVisitorExpression(DMObject dmObject, DMProc proc, DreamPath? inferredPath) {
+            _dmObject = dmObject;
+            _proc = proc;
+            _inferredPath = inferredPath;
+        }
+
+        public void VisitProcStatementExpression(DMASTProcStatementExpression statement) {
+            statement.Expression.Visit(this);
+        }
+
+
+        public void VisitConstantNull(DMASTConstantNull constant) {
+            Result = new Expressions.Null();
+        }
+
+        public void VisitConstantInteger(DMASTConstantInteger constant) {
+            Result = new Expressions.Number(constant.Value);
+        }
+
+        public void VisitConstantFloat(DMASTConstantInteger constant) {
```

Comment:
```suggestion
        public void VisitConstantFloat(DMASTConstantFloat constant) {
```

Reply:
It's kind of annoying that we can't be sure every expression visitor is implemented. Perhaps the interface should be split into 2 instead of having default implementations. (Not right now though)